arvo/clay/data-types.md

```diff
@@ -23,150 +24,180 @@ at least skimming, this so that you get a rough idea of how our state is
 
 ```hoon
-    ++  rave                                                ::  general request
-              $%  [& p=mood]                                ::  single request
-                  [| p=moat]                                ::  change range
-              ==                                            ::
+  +$  rave                                              ::  general request
+    $%  [%sing =mood]                                   ::  single request
+        [%next =mood]                                   ::  await next version
+        [%mult =mool]                                   ::  next version of any
+        [%many track=? =moat]                           ::  track range
+    ==                                                  ::
 ```
 
-This represents a subscription request for a desk. The request can be
-for either a single item in the desk or else for a range of changes on
-the desk.
+This represents a subscription request for a `desk`.
 
-### `++rove`, stored general subscription request
+A `%sing` request asks for data at single revision.
 
-```hoon
-    ++  rove  (each mood moot)                              ::  stored request
-```
+A `%next` request asks to be notified the next time there’s a change to the
+specified file.
+
+A `%mult` request asks to be notified the next time there's a change to a
+specified set of files.
 
-When we store a request, we store subscriptions with a little extra
-information so that we can determine whether new versions actually
-affect the path we're subscribed to.
+A `%many` request asks to be notified on every change in a `desk` for a range of changes (including into the future).
 
-#### `++mood:clay`, single subscription request
+### `$rove`, stored general subscription request
 
 ```hoon
-    ++  mood  ,[p=care q=case r=path]                       ::  request in desk
+    +$  rove                                                ::  stored request
+          $%  [%sing =mood]                             ::  single request
+              [%next =mood aeon=(unit aeon) =cach]      ::  next version of one
+              $:  %mult                                 ::  next version of any
+                  =mool                                 ::  original request
+                  aeon=(unit aeon)                      ::  checking for change
+                  old-cach=(map [=care =path] cach)     ::  old version
+                  new-cach=(map [=care =path] cach)     ::  new version
+              ==                                        ::
+              [%many track=? =moat lobes=(map path lobe)] ::  change range
+          ==                                            ::
 ```
 
-This represents a request for the state of the desk at a particular
-commit, specfied by `q`. `p` specifies what kind of information is
-desired, and `r` specifies the path we are requesting.
+Like a `rave` but with caches of current versions for `%next` and `%many`.
```

Comment:
I wouldn't call it a cache, since I think we need that information for correctness.  "Cache" implies it could be deleted.